packages/build-utils/test/unit.builds-and-routes-detector.test.ts

```diff
@@ -78,18 +78,64 @@ describe('Test `detectBuilders`', () => {
     expect(builders![0].use).toBe('@vercel/node');
     expect(builders![0].src).toBe('api/users.js');
     expect(builders![1].use).toBe('@vercel/static');
-    expect(builders![1].src).toBe('!{api/**,package.json}');
+    expect(builders![1].src).toBe('!{api/**,package.json,middleware.[jt]s}');
     expect(builders!.length).toBe(2);
     expect(errors).toBe(null);
   });
 
+  it('no package.json + no build + root-level "middleware.js"', async () => {
+    const files = ['middleware.js', 'index.html', 'web/middleware.js'];
+    const { builders, errors } = await detectBuilders(files);
+    console.log(builders);
```

Comment:
do you mean to have console.log statements in the tests?

Reply:
Nope. 
```suggestio
```